internal/components/datasource/schema.go

```diff
@@ -76,7 +78,7 @@ func newDataSourceJSONKernel(lin thema.Lineage) kernel.InputKernel {
 	return jdk
 }
 
-type DataSource struct {
+type DataSourceSpec struct {
```

Comment:
i know it deviates from typical k8s patterns, but IMO we should keep the name `DataSource` for this...